### ISSUE TITLE
Unresponsive GUI after searching by search terms that give a lot of results

### ISSUE THREAD

**Issue by JamesOlvertone** [zlatinb/muwire#164]
GUI becomes unresponsive after searching e.g. by /.*/ that result in a huge amount of results.

**Comment by zlatinb** [comment]
Hi,

Lots of improvements in this [CI build](https://github.com/zlatinb/muwire/actions/runs/2824584604)  Please give it a try or build from source. @searinox it would be great if you can keep an eye on this one too.

I was able to get to several hundred thousand search results with the UI still being reasonably responsive when grouping by sender.  When grouping by file I think I'm starting to hit some limits beyond my control.

Thanks

**Comment by JamesOlvertone** [comment]
Yes, great improvement of responsiveness. Even when Hashing + Searching it is responsive.

Thank you.

**Comment by Searinox** [comment]
Confirming a very big improvement in UI responsiveness!

**Comment by zlatinb** [comment]
Alright 👍 

I think I'll leave this issue open because there will always be more work to do optimizing the UI for search result display.
